src/originalBudgetTrackingApp/commonTypes.ts

```diff
@@ -1,7 +1,15 @@
-import { Transaction } from '@/originalBudgetTrackingApp/bankScraper';
+import { Config } from './configManager/configManager';
+import { Transaction } from './bankScraper';
 
 export interface EnrichedTransaction extends Transaction {
   accountNumber: string;
   category?: string;
   hash: string;
 }
+
+export interface OutputVendor {
+  name: string;
+  init?: (config: Config) => Promise<void>;
+  exportTransactions: (transactionsToCreate: EnrichedTransaction[], startDate: Date, config: Config) => Promise<any>;
```

Comment:
What about getting a subset of the whole `Config` type?

Reply:
Thats a good question. I considered making the OutputVendor generic and get the type of its own config. The problem there is that the calling code needs to know which config to fetch. This could be solved by assuming that the name of the output vendor will be the name of the output vendor config. To make that work I think we will need to make the output vendor name and enum instead of string. I will take a look

Reply:
It works, but I will need to ts ignore the code that loops over the output vendors and calls export transactions because there the ts compiler is not able to know that we are passing the correct config.

Perhaps if output vendor is a class and would have a base class that has a getVendorConfig function that will return the config for that vendor (generically typed).

Will push a solution with ts ignore and when I do the class implementation will see if the solution with the base class works.

Reply:
It's all good, but I'm afraid that I'm over-thinking. I think we can merge it, and I'm sure we will touch this code soon, so maybe the next time we will touch here we will find another solution.

In my head I always thinking about the `israeli-bank-scrapers` implementation- they have an object with the names and the fields, this is like your `enum` and `config`, and they have a `factory` to create the instance of `outputVendor` (in our case).
But I'm sure after we continue to work, if we did something wrong, we will find and fix it.